timelock-server/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosResource.java

```diff
@@ -52,13 +54,21 @@ public static PaxosResource create(String logDirectory) {
         return new PaxosResource(logDirectory, Maps.newHashMap(), Maps.newHashMap());
     }
 
-    public void addClient(String client) {
+    public void addInstrumentedClient(String client) {
         Preconditions.checkState(!paxosLearners.containsKey(client),
                 "Paxos resource already has client '%s' registered", client);
-        paxosLearners.put(client, PaxosLearnerImpl.newLearner(
-                Paths.get(logDirectory, client, PaxosTimeLockConstants.LEARNER_SUBDIRECTORY_PATH).toString()));
-        paxosAcceptors.put(client, PaxosAcceptorImpl.newAcceptor(
-                Paths.get(logDirectory, client, PaxosTimeLockConstants.ACCEPTOR_SUBDIRECTORY_PATH).toString()));
+
+        String learnerLogDir = Paths.get(logDirectory, client, PaxosTimeLockConstants.LEARNER_SUBDIRECTORY_PATH).toString();
+        PaxosLearner learner = instrument(PaxosLearner.class, PaxosLearnerImpl.newLearner(learnerLogDir), client);
+        paxosLearners.put(client, learner);
+
+        String acceptorLogDir = Paths.get(logDirectory, client, PaxosTimeLockConstants.ACCEPTOR_SUBDIRECTORY_PATH).toString();
+        PaxosAcceptor acceptor = instrument(PaxosAcceptor.class, PaxosAcceptorImpl.newAcceptor(acceptorLogDir), client);
+        paxosAcceptors.put(client, acceptor);
```

Comment:
not related to this PR, but seems odd that these maps a regular `HashMap` and not `ConcurrentHashMap`, though it looks like they're only initialized from `com.palantir.atlasdb.timelock.paxos.PaxosTimeLockServer#createPaxosBackedTimestampService`